### PR TITLE
[Feature] Support JDK 11 compiling

### LINKED ISSUE #1552: [Feature]Support JDK 11 compiling
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Hi all @apache/seatunnel-committers

According to the OpenJDK website(https://adoptopenjdk.net/support.html), there are 3 TLS versions for now

Java 8 (LTS). At Least May 2026
Java 11 (LTS). At Least Oct 2024
Java 17(LTS) TBC[1]
We should make the JDK11 compiling passed. That is the primary agenda when I started this suggestion. 
Also, If you have any concerns, please reply.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of this pull request

close #1552 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@wuchunfu Now it's ready to review, I test on my laptop, it can build success with jdk11 and submit to Flink cluster success.

**Comment by Hisoka-X** [comment]
> @wuchunfu Now it's ready to review, I test on my laptop, it can build success with jdk11 and submit to Flink cluster success.

We should generate two version of release in dist, Java 8 and Java 11.

**Comment by ruanwenjun** [comment]
@BenJFan I think maybe we only keep the current setting is Ok, the release of java 8 can execute successful on java 11.
Is there anything I haven't considerate?

**Comment by Hisoka-X** [comment]
> > 
> 
> I think maybe we only keep the current setting is Ok, the release of java 8 can execute successful on java 11. Is there anything I haven't considerate?

In Java specification, lower version code alway can run on higher version vm. The different for compile two version is higher version Java byte code have more optimizations, or new instruction.

**Comment by ruanwenjun** [comment]
> > > 
> > 
> > 
> > I think maybe we only keep the current setting is Ok, the release of java 8 can execute successful on java 11. Is there anything I haven't considerate?
> 
> In Java specification, lower version code alway can run on higher version vm. The different for compile two version is higher version Java byte code have more optimizations, or new instruction.

Yes, but I think we don't need to provide release binding with different jdk, if user want to use jdk11 to build they can build from the source by themselves, other projects seem not provide different jdk release.

**Comment by Hisoka-X** [comment]
> > > > 
> > > 
> > > 
> > > I think maybe we only keep the current setting is Ok, the release of java 8 can execute successful on java 11. Is there anything I haven't considerate?
> > 
> > 
> > In Java specification, lower version code alway can run on higher version vm. The different for compile two version is higher version Java byte code have more optimizations, or new instruction.
> 
> Yes, but I think we don't need to provide release binding with different jdk, if user want to use jdk11 to build they can build from the source by themselves, other projects seem not provide different jdk release.

You are right, most project didn't to that. It's a suggestion, maybe we can do this in the future

**Comment by CalvinKirs** [comment]
It is best to add the corresponding CI check for this

**Comment by CalvinKirs** [comment]
SkyWalking is a good example. https://github.com/apache/skywalking/

**Comment by ruanwenjun** [comment]
> It is best to add the corresponding CI check for this

Thanks for your suggestion, I added the CI with jdk11.

**Review by CalvinKirs — COMMENTED** [review]
Well done. good to see you finish this job, finally, let's explain this in the documentation.

**Comment by ruanwenjun** [comment]
> Well done. good to see you finish this job, finally, let's explain this in the documentation.

I added some information in `build.md`, please review.

File: docs/en/developement/setup.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Before we start talking about how to set up the environment, we need to do some
 have installed the following software:
 
 * [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) installed.
-* [Java](https://www.java.com/en/download/) (only JDK 8 supported by now) installed and `JAVA_HOME` set.
+* [Java](https://www.java.com/en/download/) ( JDK8/JDK11 are supported by now) installed and `JAVA_HOME` set.
 * [Scala](https://www.scala-lang.org/download/2.11.8.html) (only scala 2.11.8 supported by now) installed.
```
Comment (CalvinKirs) [inline docs/en/developement/setup.md]: You seem to have forgotten the update here

Comment (ruanwenjun) [reply docs/en/developement/setup.md]: Thanks, update the doc.

**Comment by CalvinKirs** [comment]
Hi, did you test the `spark-example`?

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**